reporting/reporter.go

```diff
@@ -115,6 +135,13 @@ func NewReporter(sender wf.Sender, application application.Tags, setters ...Opti
 		setter(r)
 	}
 
+	if !isDefaultRegistryUsed {
+		r.registry = metrics.DefaultRegistry
+		isDefaultRegistryUsed = true
+	} else {
+		r.registry = metrics.NewRegistry()
```

Comment:
Can we just make it possible to users to supply the registry to be used. i.e have a setter func that will override the registry the reporter uses. Let it default to metrics.DefaultRegistry as before if not overriden.